examples/BUILD.bazel

```diff
@@ -19,7 +23,11 @@ bazel_env(
         "nodejs": "@nodejs_toolchains//:resolved_toolchain",
         "rust": "@rules_rust//rust/toolchain:current_rust_toolchain",
         "rules_python_docs": "@rules_python_docs",
-    },
+    } | (
+        {
+            "go": "@rules_go//go:toolchain",
+        } if _TOOLCHAIN_TYPES_SUPPORTED else {}
+    ),
```

Comment:
Nit: Imo the rule should does this automatically underneath.

This is an `examples` BUILD file, it should mostly contain copy-paste-able snippets for the user to mimic. We don't want the user having to mess with `bazel_features` and complicated if/else like this.